flake.nix

```diff
@@ -3,230 +3,228 @@
-          [
-            pkgs.terraform_0_15
-            self.packages.${system}.terranix
-            pkgs.treefmt
-            pkgs.nixpkgs-fmt
-            pkgs.shfmt
-            pkgs.shellcheck
-            pkgs.nodePackages.prettier
-          ];
-      };
-      # TODO: Legacy attribute, drop soon
-      devShell = self.devShells.${system}.default;
-
-      # nix run
-      apps.default = self.apps.${system}.test;
-      # TODO: Legacy attribute, drop soon
-      defaultApp = self.apps.${system}.default;
-      # nix run ".#test"
-      apps.test =
-        let
-          tests = import ./tests/test.nix {
-            inherit nixpkgs;
-            inherit pkgs;
-            inherit (pkgs) lib;
-            terranix = self.packages.${system}.terranix;
+  outputs = inputs @ { self, flake-parts, nixpkgs, ... }:
+
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      imports = [
+      ];
+
+      systems = import inputs.systems;
+
+      perSystem =
+        { config
+        , self'
+        , pkgs
+        , system
+        , ...
+        }: {
+          packages = rec {
+            default = terranix;
+
+            terranix = pkgs.callPackage ./default.nix {
+              nix = pkgs.nixUnstable;
+            };
+
+            inherit (pkgs.callPackage ./doc/default.nix { }) manPages;
           };
-          testFile = pkgs.writeText "test" ''
-            load '${bats-support}/load.bash'
-            load '${bats-assert}/load.bash'
-            ${pkgs.lib.concatStringsSep "\n" tests}
-          '';
-        in
-        {
-          type = "app";
-          program = toString (pkgs.writeShellScript "test" ''
-            set -e
-            echo "running terranix tests" | ${pkgs.boxes}/bin/boxes -d ian_jones -a c
-            #cat ${testFile}
-            ${pkgs.bats}/bin/bats ${testFile}
-          '');
-        };
-      # nix run ".#docs"
-      apps.doc = self.apps.${system}.docs;
-      apps.docs = {
-        type = "app";
-        program = toString (pkgs.writeShellScript "docs" ''
-          set -e
-          export PATH=${pkgs.pandoc}/bin:$PATH
-          ${pkgs.gnumake}/bin/make --always-make --directory=doc
-          nix build ".#manpages"
-          cp -r result/share .
-          chmod -R 755 ./share
-          rm result
-        '');
-      };
 
-      formatter = pkgs.treefmt;
-    })) // {
-
-      # terraformConfiguration ast, if you want to run
-      # terranix in the repl.
-      lib.terranixConfigurationAst =
-        { system ? ""
-        , pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages
-        , extraArgs ? { }
-        , modules ? [ ]
-        , strip_nulls ? true
-        }:
-        import ./core/default.nix {
-          inherit pkgs extraArgs strip_nulls;
-          terranix_config.imports = modules;
-        };
+          devShells.default = pkgs.mkShell {
+            buildInputs = with pkgs;
+              [
+                terraform_0_15
+                self'.packages.terranix
```

Comment:
This can also be `config.packages.terranix`

Reply:
> This can also be `config.packages.terranix`

I interpreted https://flake.parts/cheat-sheet#in-persystem as `self'` being the preferred access point for outputs that support it due to the potential future caching.